src/eve-common/utils/EvEMath.cpp

```diff
@@ -65,14 +65,18 @@ int32 EvEMath::RAM::ProductionTime(uint32 BaseTime, float bpProductivityModifier
     return (BaseTime * effModifier * TimeModifier);
 }
 
-int32 EvEMath::RAM::ME_ResearchTime(uint32 BaseTime, uint8 MetallurgyLevel, float SlotModifier/*1*/, float ImplantModifier/*1*/ )
+const int LEVELMODIFIERS [11] = { 0, 105, 250, 595, 1414, 3360, 8000, 19000, 45255, 107700, 256000 }; // This probably needs to go to a more appropriate location
```

Comment:
This should go into it's own function with a bounds check to ensure we do not read out of the array as the "Runs" parameter comes directly from the client's call, which is easily forged.

Reply:
Yes that right, in its current state it works well up to level 10 and then for 11 and above it returns zero which the client interprets as an instantaneous job.
My intention to fix this is to add a check earlier on that feeds back to the client (I'm still working out how to do this) when the current level + runs > 10 which will fix the current issue of being able to enter too many runs for a research job and by extension solve the boundary issue at the same time.

Another thing I will also need to make allowances for is that wile ME goes up by 1 to a maximum of 10, PE goes up by 2 up to a maximum of 20 so I'll need to ensure the code can handle that.

Thoughts?

Reply:
It's usually a good idea to have a bounds check even if you validate the client's data, this way you ensure that if someone forgets to do it that your app doesn't crash or read garbage data, which might or might not be what you expect. With the changes it looks good to me now. Nice work ;)